plugins/org.eclipse.xtend.core/src/org/eclipse/xtend/core/javaconverter/JavaASTFlattener.xtend

```diff
@@ -401,12 +401,23 @@ class JavaASTFlattener extends ASTVisitor {
 			body.accept(this)
 			prev = body
 		}
+		if (it.root instanceof CompilationUnit) {
+			val cu = it.root as CompilationUnit
+			cu.unAssignedComments.forEach [
+				accept(this)
+				assignedComments.add(it)
+			]
+		}
 		decreaseIndent
 		appendLineWrapToBuffer
 		appendToBuffer("}")
 		return false
 	}
 
+	def private Iterable<Comment> unAssignedComments(CompilationUnit cu) {
```

Comment:
this should probably be called unassignedComments (lowercase a) or getUnassignedComments